magic-nix-cache/src/main.rs

```diff
@@ -154,6 +154,15 @@ impl Args {
     }
 }
 
+fn default_nix_conf() -> PathBuf {
+    xdg::BaseDirectories::new()
+        .with_context(|| "identifying XDG base directories")
+        .expect(
```

Comment:
yes yes we expect, but imho its fine... it happens _immediately_ at startup in a rare scenario:

```
Returns the value of the ‘HOME’ environment variable if it is set (including to an empty string).
Otherwise, it tries to determine the home directory by invoking the getpwuid_r function using the UID of the current user. An empty home directory field returned from the getpwuid_r function is considered to be a valid value.
Returns None if the current user has no entry in the /etc/passwd file.
```